src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -149,57 +149,46 @@ MulticopterAttitudeControl::generate_attitude_setpoint(const Quatf &q, float dt,
 		v *= _man_tilt_max / v_norm;
 	}
 
-	Quatf q_sp_rpy = AxisAnglef(v(0), v(1), 0.f);
-	Eulerf euler_sp = q_sp_rpy;
-	attitude_setpoint.roll_body = euler_sp(0);
-	attitude_setpoint.pitch_body = euler_sp(1);
+	Quatf q_sp_rp = AxisAnglef(v(0), v(1), 0.f);
 	// The axis angle can change the yaw as well (noticeable at higher tilt angles).
 	// This is the formula by how much the yaw changes:
 	//   let a := tilt angle, b := atan(y/x) (direction of maximum tilt)
 	//   yaw = atan(-2 * sin(b) * cos(b) * sin^2(a/2) / (1 - 2 * cos^2(b) * sin^2(a/2))).
-	attitude_setpoint.yaw_body = _man_yaw_sp + euler_sp(2);
+	const Vector3f z_unit(0.f, 0.f, 1.f);
+	Quatf q_sp_yaw = AxisAnglef(z_unit, _man_yaw_sp);
```

Comment:
could be replaced by `q_yaw{cosf(_man_yaw_sp / 2.f), 0.f, 0.f, sinf(_man_yaw_sp / 2.f)}`

Reply:
Let's use that then as it's also a single line.